extensions/guacamole-auth-restrict/src/main/resources/controllers/timeRestrictionFieldController.js

```diff
@@ -0,0 +1,222 @@
+    // const TimeRestrictionEntry = $injector.get('TimeRestrictionEntry');
+    
+    /**
+     * The restrictions, as objects, that are used by the HTML template to
+     * present the restrictions to the user via the web interface.
+     */
+    $scope.restrictions = [];
+
+    /**
+     * Map of weekday identifier to display name.
+     */
+    $scope.weekDays = [
+        { id : 1, day : 'Monday' },
+        { id : 2, day : 'Tuesday' },
+        { id : 3, day : 'Wednesday' },
+        { id : 4, day : 'Thursday' },
+        { id : 5, day : 'Friday' },
+        { id : 6, day : 'Saturday' },
+        { id : 7, day : 'Sunday' }
+    ];
+    
+    /**
+     * Remove the current entry from the list.
+     * 
+     * @param {type} entry
+     *     A restriction entry.
+     */
+    $scope.removeEntry = function removeEntry(entry) {
+        if (entry === null || entry.$$hashKey === '') {
+            return;
+        }
+        for (let i = 0; i < $scope.restrictions.length; i++) {
+            if ($scope.restrictions[i].$$hashKey === entry.$$hashKey) {
+                $scope.restrictions.splice(i,1);
+                return;
+            }
+        }
+    };
+    
+    /**
+     * Add an empty entry to the restriction list.
+     */
+    $scope.addEntry = function addEntry() {
+        $scope.restrictions.push({});
+    };
+    
+
+    /**
+     * Parse the provided string into an array containing the objects that
+     * represent each of entries that can then be displayed as a more
+     * user-friendly field.
+     * 
+     * @param {String} restrString
+     *     The string that contains the restrictions, un-parsed and as stored
+     *     in the underlying field.
+     *     
+     * @returns {Array|Object}
+     *     An array of objects that represents each of the entries as parsed
+     *     out of the string field, and which can be interpreted by the
+     *     AngularJS field for display.
+     */
+    var parseRestrictions = function parseRestrictions(restrString) {
+        
+        var restrictions = [];
+        
+        // If the string is null or empty, just return an empty array
+        if (restrString === null || restrString === "")
+            return restrictions;
+        
+        // Set up the RegEx and split the string using the separator.
+        const restrictionRegex = new RegExp('^((?:[1-7]\-)?(?:[1-7]))?(?::((?:[01][0-9]|2[0-3])[0-5][0-9])\-((?:[01][0-9]|2[0-3])[0-5][0-9]))$');
+        var restrArray = restrString.split(";");
+
+        // Loop through split string and process each item
+        for (let i = 0; i < restrArray.length; i++) {
+            if (restrictionRegex.test(restrArray[i])) {
+                var currArray = restrArray[i].match(restrictionRegex);
+                var restrDays = currArray[1].split("-");
+                
+                if (restrDays.length > 1) {
+                    for (let j = restrDays[0]; j <= restrDays[1]; j++) {
+                        var entry = {};
```

Comment:
I'd consider defining a type for these restriction objects to make it clearer what exactly the fields are, and what the format of each field should be.

Reply:
Yeah - I actually had it that way, in fact, the code is still there. I was just running into issues with it and temporarily removed it, then forgot to go back and put it back...

Reply:
Fixed via rebase.